From 5837c135663d8ec144dc14e77c0e5c203d82a803 Mon Sep 17 00:00:00 2001
From: Jiajian Wu <jair.wu@rock-chips.com>
Date: Mon, 18 Mar 2024 19:27:24 +0800
Subject: [PATCH 02/12] feat(draw): add LV_DRAW_BUF_PITCH_ALIGN

Signed-off-by: Jiajian Wu <jair.wu@rock-chips.com>
---
 src/draw/lv_draw_buf.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/draw/lv_draw_buf.c b/src/draw/lv_draw_buf.c
index bd7f740..f070e65 100644
--- a/src/draw/lv_draw_buf.c
+++ b/src/draw/lv_draw_buf.c
@@ -485,6 +485,10 @@ static void * buf_align(void * buf, lv_color_format_t color_format)
 static uint32_t width_to_stride(uint32_t w, lv_color_format_t color_format)
 {
     uint32_t width_byte;
+
+#ifdef LV_DRAW_BUF_PITCH_ALIGN
+    w = (w + LV_DRAW_BUF_PITCH_ALIGN - 1) & ~(LV_DRAW_BUF_PITCH_ALIGN - 1);
+#endif
     width_byte = w * lv_color_format_get_bpp(color_format);
     width_byte = (width_byte + 7) >> 3; /*Round up*/
     return (width_byte + LV_DRAW_BUF_STRIDE_ALIGN - 1) & ~(LV_DRAW_BUF_STRIDE_ALIGN - 1);
-- 
2.25.1

