From 2158e5c75aa041fa8e8e13e921ed081946846c82 Mon Sep 17 00:00:00 2001
From: Jiajian Wu <jair.wu@rock-chips.com>
Date: Thu, 24 Oct 2024 19:11:20 +0800
Subject: [PATCH 06/22] fix(cmake): add drm include path

Signed-off-by: Jiajian Wu <jair.wu@rock-chips.com>
---
 env_support/cmake/custom.cmake | 1 +
 1 file changed, 1 insertion(+)

diff --git a/env_support/cmake/custom.cmake b/env_support/cmake/custom.cmake
index d0e82b713..b366a4680 100644
--- a/env_support/cmake/custom.cmake
+++ b/env_support/cmake/custom.cmake
@@ -44,6 +44,7 @@ endif()
 target_include_directories(lvgl SYSTEM PUBLIC ${LVGL_ROOT_DIR} ${LV_CONF_DIR})
 
 target_include_directories(lvgl SYSTEM PUBLIC ${CMAKE_SYSROOT}/usr/include/freetype2)
+target_include_directories(lvgl SYSTEM PUBLIC ${CMAKE_SYSROOT}/usr/include/drm)
 
 if(NOT LV_CONF_BUILD_DISABLE_THORVG_INTERNAL)
     add_library(lvgl_thorvg ${THORVG_SOURCES})
-- 
2.25.1

