From 5612b4f3c1a09762e1e1a2c73c44e5fcf8b98182 Mon Sep 17 00:00:00 2001
From: ymh-masuda <160470322+ymh-masuda@users.noreply.github.com>
Date: Fri, 24 May 2024 17:01:01 +0900
Subject: [PATCH 07/35] fix(roller): Avoid divided-by-zero during draw event
 (#6259)

---
 src/widgets/lv_roller.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/widgets/lv_roller.c b/src/widgets/lv_roller.c
index bda15ad51..bcfc3142f 100644
--- a/src/widgets/lv_roller.c
+++ b/src/widgets/lv_roller.c
@@ -488,7 +488,10 @@ static void draw_main(lv_event_t * e)
 
             /*Proportional position from the middle line.
              *Will be 0 for the first option, and 1 for the last option (upscaled by << 14)*/
-            label_y_prop = (label_y_prop << 14) / (lv_obj_get_height(label) - normal_label_font->line_height);
+            lv_coord_t remain_h = lv_obj_get_height(label) - normal_label_font->line_height;
+            if(remain_h > 0) {
+                label_y_prop = (label_y_prop << 14) / remain_h;
+            }
 
             /*We don't want the selected label start and end exactly where the normal label is as
              *a larger font won't centered on selected area.*/
-- 
2.25.1

